File: spring/order-service/src/main/java/com/microsoft/gbb/reddog/orderservice/model/OrderItemSummary.java
package com.microsoft.gbb.reddog.orderservice.model;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * The type Order item summary.
 */
public class OrderItemSummary {

    @JsonProperty("unitPrice")
    private double unitPrice;

    @JsonProperty("quantity")
    private int quantity;

    @JsonProperty("productId")
    private int productId;

    @JsonProperty("unitCost")
    private double unitCost;

    @JsonProperty("imageUrl")
    private String imageUrl;

    @JsonProperty("productName")
    private String productName;

    /**
     * Sets unit price.
     *
     * @param unitPrice the unit price
     */
    public void setUnitPrice(double unitPrice) {
        this.unitPrice = unitPrice;
    }

    /**
     * Gets unit price.
     *
     * @return the unit price
     */
    public double getUnitPrice() {
        return unitPrice;
    }

    /**
     * Sets quantity.
     *
     * @param quantity the quantity
     */
    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    /**
     * Gets quantity.
     *
     * @return the quantity
     */
    public int getQuantity() {
        return quantity;
    }

    /**
     * Sets product id.
     *
     * @param productId the product id
     */
    public void setProductId(int productId) {
        this.productId = productId;
    }

    /**
     * Gets product id.
     *
     * @return the product id
     */
    public int getProductId() {
        return productId;
    }

    /**
     * Sets unit cost.
     *
     * @param unitCost the unit cost
     */
    public void setUnitCost(double unitCost) {
        this.unitCost = unitCost;
    }

    /**
     * Gets unit cost.
     *
     * @return the unit cost
     */
    public double getUnitCost() {
        return unitCost;
    }

    /**
     * Sets image url.
     *
     * @param imageUrl the image url
     */
    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    /**
     * Gets image url.
     *
     * @return the image url
     */
    public String getImageUrl() {
        return imageUrl;
    }

    /**
     * Sets product name.
     *
     * @param productName the product name
     */
    public void setProductName(String productName) {
        this.productName = productName;
    }

    /**
     * Gets product name.
     *
     * @return the product name
     */
    public String getProductName() {
        return productName;
    }
}

File: spring/accounting-service/src/main/java/com/microsoft/gbb/reddog/accountingservice/entities/OrderItemSummary.kt
package com.microsoft.gbb.reddog.accountingservice.entities

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.Builder
import lombok.Getter
import lombok.Setter

/**
 * The type Order item summary.
 */
@Getter
@Setter
@Builder
class OrderItemSummary {
    @JsonProperty("unitPrice")
    private val unitPrice = 0.0

    @JsonProperty("quantity")
    private val quantity = 0

    @JsonProperty("productId")
    private val productId = 0

    @JsonProperty("unitCost")
    private val unitCost = 0.0

    @JsonProperty("imageUrl")
    private val imageUrl: String? = null

    @JsonProperty("productName")
    private val productName: String? = null
    override fun toString(): String {
        return "OrderItemSummary{" +
                "unitPrice=" + unitPrice +
                ", quantity=" + quantity +
                ", productId=" + productId +
                ", unitCost=" + unitCost +
                ", imageUrl='" + imageUrl + '\'' +
                ", productName='" + productName + '\'' +
                '}'
    }
}